homeassistant/helpers/entity_platform.py

```diff
@@ -269,10 +272,22 @@ def add_entities(self, new_entities, update_before_add=False):
             else:
                 config_entry_id = None
 
+            if entity.device is not None:
+                device = device_registry.async_get_or_create(
+                    entity.device['unique_id'], entity.device['serial'],
+                    entity.device['manufacturer'], entity.device['model'],
+                    entity.device['connection'],
+                    sw_version=entity.device.get('sw_version'),
+                    config_entry_id=config_entry_id)
+                device_id = device.unique_id
```

Comment:
Should be `device.id`, we need to generate our own IDs for devices and not trust developers.